### PR TITLE
Add new edit_editions permission

### PR DESCRIPTION
**Pull request by aug24**
## What's changed?
Check for a new permission ("edit_editions") before permitting access to the Editions UI.

## Implementation notes
The majority of this work is in the Permissions application.  This work effectively just changes the Permission check for all the routes used by Editions.

### General
- [ ] 🤖 Relevant tests added
- [X ] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [ ] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [ ] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

**Comment by akash1810** [comment]
Do we want to restrict read access, or just write access? In other Tools we allow anyone to read (provided you have a Guardian email) and permission editing/publishing.

**Comment by aug24** [comment]
> Do we want to restrict read access, or just write access? In other Tools we allow anyone to read (provided you have a Guardian email) and permission editing/publishing.

The ticket merely says separate permissions.  All these routes had the same permission requirement before, which was "edit_editorial_fronts".

File: app/permissions/CollectionPermissions.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package permissions
+
+import com.gu.facia.client.models.ConfigJson
+
+object CollectionPermissions {
+  def getFrontsPermissionsPriorityByCollectionId(config: Option[ConfigJson], id: String): Set[PermissionsPriority] = config match {
+    case None => Set.empty[PermissionsPriority] // if there are no fronts in config, there can be no permissions....?
+    case Some(config) => {
+      config.fronts.values.flatMap(front =>
```
Comment (sihil) [inline app/permissions/CollectionPermissions.scala]: I'd use `collect` here.

ie.
```
config.fronts.values.collect {
  case front if front.collections.contains(id) => PermissionsPriority.priorityOptionToPermissionPriority(front.priority)
}
```

File: app/permissions/CollectionPermissions.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        if (front.collections.contains(id))
+          // Will need the permission for this front
+          PermissionsPriority.priorityOptionToPermissionPriority(front.priority)
+        else
+          None
+    ) match {
```
Comment (sihil) [inline app/permissions/CollectionPermissions.scala]: The indenting here is wrong. ⇥

File: app/permissions/CollectionPermissions.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        // If this collection id is not present in any of the fronts, then it must belong to an edition issue
+        case Nil => Set[PermissionsPriority](EditionsPermission)
+        case permissionsFromFronts => permissionsFromFronts.toSet
+      }
+    }
+  }
+
+}
```
Comment (sihil) [inline app/permissions/CollectionPermissions.scala]: This is not particularly elegant but I've not managed to improve it in the time I've looked at it. Perhaps breaking it into a couple of functions might help. Although the whole concept is pretty confusing to begin with TBH.

File: app/permissions/PermissionsPriority.scala [hunk 4]
```diff
@@ -5,15 +5,20 @@ case object EditorialPermission extends PermissionsPriority
 case object CommercialPermission extends PermissionsPriority
 case object EmailPermission extends PermissionsPriority
 case object TrainingPermission extends PermissionsPriority
+case object EditionsPermission extends PermissionsPriority
 
 
 object PermissionsPriority {
+  def priorityOptionToPermissionPriority(os: Option[String]): Option[PermissionsPriority] = {
+    stringToPermissionPriority(os.getOrElse("editorial"))
```
Comment (sihil) [inline app/permissions/PermissionsPriority.scala]: weirdly I'd prefer:
```suggestion
    os.flatMap(stringToPermissionPriority).orElse(Some(EditorialPermission))
```

File: app/permissions/PermissionsActionCheck.scala [hunk 5]
```diff
@@ -74,15 +75,30 @@ class EditEditorialFrontsPermissionCheck(val acl: Acl)(implicit ec: ExecutionCon
...
   }
 }
 
+class EditEditionsPermissionCheck(client: PermissionsProvider)(implicit ec: ExecutionContext) extends PermissionActionFilter {
+  val executionContext = ec
+  val restrictedAction = "edit editions"
+  val testAccess: String => Authorization = (email: String) => {
+    val hasPermission = client.hasPermission(Permissions.EditEditions, email)
+    if(hasPermission) { AccessGranted } else { AccessDenied }
+  }
```
Comment (sihil) [inline app/permissions/PermissionsActionCheck.scala]: Perhaps extract this out like `testUser` above.

File: client-v2/src/components/Home.tsx [hunk 6]
```diff
@@ -16,7 +16,7 @@ const Home = () => (
     <h3>Manage editions</h3>
     <ul>
       <li>
-        <Link to={`/manage-editions/daily-edition`}>Daily edition</Link>
+        <Link to={`/manage-editions/daily-edition`}>Daily Edition</Link>
```
Comment (sihil) [inline client-v2/src/components/Home.tsx]: This is a hardcoded list 😞
Room for improvement.

Comment (aug24) [reply client-v2/src/components/Home.tsx]: Fixed in the training PR.

File: app/permissions/PermissionsPriority.scala (hunk 4 above)
Comment (aug24) [reply app/permissions/PermissionsPriority.scala]: Done

File: app/permissions/CollectionPermissions.scala (hunk 1 above)
Comment (aug24) [reply app/permissions/CollectionPermissions.scala]: Done.

File: app/permissions/CollectionPermissions.scala (hunk 2 above)
Comment (aug24) [reply app/permissions/CollectionPermissions.scala]: Done.

File: app/permissions/CollectionPermissions.scala [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package permissions
+
+import com.gu.facia.client.models.ConfigJson
+
+object CollectionPermissions {
```
[the remark this answers is not in the document]
Comment (aug24) [reply app/permissions/CollectionPermissions.scala]: Done.

File: app/permissions/PermissionsActionCheck.scala (hunk 5 above)
Comment (aug24) [reply app/permissions/PermissionsActionCheck.scala]: Done.

**Pull request merged**